### ISSUE TITLE
Subscription support

### ISSUE THREAD

**Issue by jcornaz** [oolong-kt/oolong#104]
**Is your feature request related to a problem? Please describe.**

I want to subscribe to some events, that might be external to my system. Example:

* Get an event every X seconds
* Subscribe to events from a websocket

**Describe the solution you'd like**

Since the present library is inspired from elm, I'd find natural to imitate elm's subscription system (in [Browser.element](https://package.elm-lang.org/packages/elm/browser/latest/Browser#element)).

That could be translated in Kotlin like this:

```kotlin
data class Model(val ticks: Int = 0)
sealed class Msg {
  object Tick : Msg()
}

val subscribe = Subscribe<Model, Msg> { model ->
  every(10.seconds).map { Msg.Tick }
}

// ...

val dispose = Oolong.runtime(
    init,
    update,
    suscribe,
    view,
    render
)
```

Of course this is only a draft example, and any variation of the API would do.

**Describe alternatives you've considered**

I think I could start a coroutine from `update` that will call `dispatch` to fire the events.

That would however require me to keep an instance of the Job in my model, so that I can cancel it later.

It'd be nice if `Oolong` could do this housekeeping for me.

**Comment by jcornaz** [comment]
Actually, after playing a little bit, I think using `disposableEffect` to start a long running job that fire messages is not so bad.

Maybe all we need is only documentation on how create/cancel a subscriptions using `disposableEffect`.

**Comment by pardom** [comment]
Yes, I agree that we need better documentation, recipes, and examples. You're right on track with using `disposableEffect` for subscriptions that you want to cancel, though. For reference, here's an example of a tick subscription which is very similar to what you already have:

https://github.com/oolong-kt/samples/blob/master/samples/time/core/src/commonMain/kotlin/time/Time.kt#L25

I'll add this to the [documentation project](https://github.com/oolong-kt/oolong/projects/2). Thanks for the contribution!
